quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java

```diff
@@ -744,13 +982,18 @@ private static IAnnotation getAnnotation(IAnnotatable annotatable, String annota
 		}
 		IAnnotation[] annotations = annotatable.getAnnotations();
 		for (IAnnotation annotation : annotations) {
-			if (annotationName.equals(annotation.getElementName())) {
+			if (isMatchAnnotation(annotation, annotationName)) {
 				return annotation;
 			}
 		}
 		return null;
 	}
 
+	private static boolean isMatchAnnotation(IAnnotation annotation, String annotationName) {
+		return annotationName.equals(annotation.getElementName())

```

Comment:
annotationName.equals(annotation.getElementName()) is already covered by annotationName.endsWith(annotation.getElementName())

Reply:
No it's not the same thing. annotation.getElementName() can have 2 values:

 * package + annotation name (it's the case when annotation comes from JAR)
 * just annotation name (it's the case when annotation comes from Java sources)